src/mbgl/layout/symbol_layout.cpp

```diff
@@ -358,6 +359,67 @@ std::array<float, 2> SymbolLayout::evaluateVariableOffset(style::SymbolAnchorTyp
     return result;
 }
 
+std::optional<VariableAnchorOffsetCollection> SymbolLayout::getTextVariableAnchorOffset(const SymbolFeature& feature) {
+    std::optional<VariableAnchorOffsetCollection> result;
+
+    // If style specifies text-variable-anchor-offset, just return it
+    if (!textVariableAnchorOffset.isUndefined()) {
+        std::vector<AnchorOffsetPair> anchorOffsets;
+        auto variableAnchorOffset = layout->evaluate<TextVariableAnchorOffset>(zoom, feature, canonicalID);
+        if (!variableAnchorOffset.empty()) {
+            // Convert offsets from EM to PX, and apply baseline shift
+            for (const auto& anchorOffset : variableAnchorOffset) {
```

Comment:
And here too vector::reserve() might help. Check other places where we know size ahead of time? I expect those arrays to be small, but it's still a good practice to do where we can.